src/dtree.ml

```diff
@@ -0,0 +1,648 @@
+(** This module provides functions to compile rewrite rules to decision trees
+    in order to pattern match the rules efficiently.  The method is based on
+    Luc Maranget's {i Compiling Pattern Matching to Good Decision Trees},
+    {{:10.1145/1411304.1411311}DOI}. *)
+open Terms
+open Extra
+module Sub = Basics.Subterm
+module SubMap = Basics.SubtMap
```

Comment:
Why not using `open Basics`?

Reply:
I rather changed calls to `SubMap` to calls to `SubtMap`